web/handlers.go

```diff
@@ -177,15 +178,52 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("X-Frame-Options", "SAMEORIGIN")
 
 		// Add unsafe-eval to the content security policy for faster source maps in development mode
-		devCSP := ""
+		devCSPMap := make(map[string]bool)
 		if model.BuildNumber == "dev" {
-			devCSP += " 'unsafe-eval'"
+			devCSPMap["unsafe-eval"] = true
 		}
 
 		// Add unsafe-inline to unlock extensions like React & Redux DevTools in Firefox
 		// see https://github.com/reduxjs/redux-devtools/issues/380
 		if model.BuildNumber == "dev" {
-			devCSP += " 'unsafe-inline'"
+			devCSPMap["unsafe-inline"] = true
+		}
+
+		// Add supported flags for debugging during development, even if not on a dev build.
+		for _, devFlagKeyValueStr := range strings.Split(*c.App.Config().ServiceSettings.DeveloperFlags, ",") {
+			devFlagKeyValueSplit := strings.SplitN(devFlagKeyValueStr, "=", 2)
+			if len(devFlagKeyValueSplit) != 2 {
+				c.Logger.Warn("Unable to parse developer flag", mlog.String("developer_flag", devFlagKeyValueStr))
+				continue
+			}
+			devFlagKey := devFlagKeyValueSplit[0]
+			devFlagValue := devFlagKeyValueSplit[1]
+
+			// Ignore disabled keys
+			if devFlagValue != "true" {
+				continue
+			}
```

Comment:
This is not correct. They keys can have any value. What values can a key have is decided by the key.

In the switch-case, check for the value, and if it's true, then set the map entry. You might also want to handle quotes. But it's not necessary for this PR. If you want to leave it, then just add a comment that quotes are not handled.

Reply:
@agnivade you mean we should support flag other than `unsafe-eval` and `unsafe-inline` ?

Reply:
We could definitely support more developer flags in other places, at which point we could refactor the necessary code to parse thereafter.

But in the context of `web/handlers.go`, I think there's only `unsafe-eval` and `unsafe-inline`, and we shouldn't pre-emptively add more complexity here. Unless we also want to support a dynamic type of CSP directive?

Reply:
Got it, I don't think we would add any more complexity than a few more lines. We can leave quote handling out of this. But let's move the parsing logic to a separate function.